### ISSUE TITLE
Derpibooru 3+ tags search still fails

### ISSUE THREAD

**Issue by Fillyosopher** [Bionus/imgbrd-grabber#863]
### What steps will reproduce the problem?
- Add the API_KEY to the password field in derpibooru's options
- Input a search with 3 or more tags formatted for proper derpibooru searching.  For example: "faves.gte:250 AND upvotes.gte:350 AND explicit AND animation"
- Alternate option, input 3 or more tags formatted for proper derpibooru searching while attempting to for + separation.  For example: "faves.gte:250+AND+upvotes.gte:350+AND+explicit+AND+animation"

### What is the expected behavior? What do you get instead?
I expect that Grabber would load the page
> https://derpibooru.org/search.json?key={api_key}&page=1&q=faves.gte%3A250+AND+upvotes.gte%3A350+AND+explicit+AND+animation&nocomments=1&nofav=1
on one of those two inputs.  This page has an image list as you'd expect.

What happens instead is that Grabber loads
> https://derpibooru.org/search.json?key={api_key} &page=1&q=faves.gte%3A250 AND upvotes.gte%3A350 AND explicit AND animation&nocomments=1&nofav=1

or it loads
> https://derpibooru.org/search.json?key={api_key} &page=1&q=faves.gte%3A250%2BAND%2Bupvotes.gte%3A350%2BAND%2Bexplicit%2BAND%2Banimation&nocomments=1&nofav=1
Neither of these pages have any images, so it displays nothing.  

### Please provide any additional information below
**TL;DR: The difference between what it does and should do is that the separate tags should be + separated, but are being space separated!  Trying to manually work around this is impossible since + characters are parsed to %2B.  Changing to + separation at some point for derpibooru should fix this.**

As an aside, mainwindow.cpp line 2380, inside mainWindows::loadTag, possibly change from 
`	if (tag.startsWith("http://"))` 
to
`	if (tag.startsWith("http://") || tag.startsWith("https://"))` ?

HUGE thank you for the new URL downloader! 👍  I was afraid I'd have to wait another few weeks for this to be solved, but it's allowing me to bypass this problem entirely by making the URLs I need by hand.  Thank you thank you thank you.  Because of that, I figure this issue is low priority, unless other people complain.

**Comment by Fillyosopher** [comment]
Actually, I tried single tag searchs and tag

**Comment by Fillyosopher** [comment]
accidental close, apologies

**Comment by Bionus** [comment]
It's working find for me though. Isn't it just because you added a space at the end of your API key? `+` and ` ` should be interchangeable in encoded URLs, so much that some browsers even edit them out or don't escape them at all.

> As an aside, mainwindow.cpp line 2380, inside mainWindows::loadTag, possibly change from
> if (tag.startsWith("http://"))
> to
> if (tag.startsWith("http://") || tag.startsWith("https://")) ?

Done, were tags from HTTPS source not opening on click? 😱

> HUGE thank you for the new URL downloader! 👍 I was afraid I'd have to wait another few weeks for this to be solved, but it's allowing me to bypass this problem entirely by making the URLs I need by hand. Thank you thank you thank you.

Yeah, that's exactly why I did it. As well as occasional requests for downloading some source-specific pages, for which I don't really feel like adding UIs yet. I may add a new special "URL" tab though (in addition to the current "tags" and "pool" tabs).

**Comment by Bionus** [comment]
Bump? Looks like your issue was a space at the end of your API key (from what you copy-pasted in your post).

Can I consider the issue solved? :smile:
